core/src/test/java/de/danoeh/antennapod/core/util/DateUtilsTest.java

```diff
@@ -12,16 +12,17 @@
 
 /**
  * Unit test for {@link DateUtils}.
- *
- * Note: It NEEDS to be run in android devices, i.e., it cannot be run in standard JDK, because
- * the test invokes some android platform-specific behavior in the underlying
- * {@link java.text.SimpleDateFormat} used by {@link DateUtils}.
- *
  */
-@SmallTest
+@SuppressWarnings("ConstantConditions")
 public class DateUtilsTest {
+
+    @Before
+    public void setUp() {
+        DateUtils.dateFormatParser.setTimeZone(DateUtils.defaultTimezone);
```

Comment:
Why is this needed? Isn't this done in the `static` block automatically?

Reply:
Sorry, I forgot to comment on this. `SimpleDateFormat.parse()` modifies the configured timezone if the date string contains a timezone, e.g. `"Sat, 28 Mar 2015 01:31 EST"`. As we reuse a global `SimpleDateFormat` instance this leads to side effects and breaks some unit tests. So I had to add the `setUp()` method.

**Why wasn't that a problem before?**

Android has a modified version of `SimpleDateFormat` that avoids that side effect. See the comment in the Android source code snipped below:

```java
public Date parse(String text, ParsePosition pos) {
    // BEGIN Android-changed: extract parseInternal() and avoid modifying timezone during parse.
    // Make sure the timezone associated with this dateformat instance (set via
    // {@code setTimeZone} isn't change as a side-effect of parsing a date.
    final TimeZone tz = getTimeZone();
    try {
        return parseInternal(text, pos);
    } finally {
        setTimeZone(tz);
    }
}
```
----

BTW: The usage of a global `SimpleDateFormat` instance isn't thread-safe and thus error-prone. I'm aware that the current implementation intends to speed up date parsing but on the other side it parses a single date string with up to 29 patterns. Maybe we could speed up date parsing even more by an own date parser implementation or find an appropriate open source implementation.

Reply:
Oh, right. This was reverted in a bug fix release for 2.1.x (#4703 and the error in #4648) but I did not merge `master` into `develop` yet. Will do that today.

Reply:
Great to hear. So I will wait for the merge, remove the `setTimeZone()` call and test again.

Reply:
It's already merged :)

Reply:
Perfect, I updated the code. (This time it looks like I didn't mess up the commit history :grin:)

Reply:
I just fixed the `Illegal connection pointer` thing in #4828. Could you rebase again to make the tests apss, please? :see_no_evil:

Reply:
Done.